cmd/tsbs_generate_queries/databases/timescaledb/devops_test.go

```diff
@@ -152,3 +154,333 @@ func TestDevopsGetSelectClausesAggMetrics(t *testing.T) {
+          avg(usage_user) as mean_usage_user
+          FROM cpu
+          WHERE time >= '1970-01-01 00:16:22.646325 +0000' AND time < '1970-01-01 12:16:22.646325 +0000'
+          GROUP BY hour, tags_id
+        )
+        SELECT hour, hostname, mean_usage_user
+        FROM cpu_avg
+        
+        ORDER BY hour, hostname`,
+		},
+		{
+			desc:               "use JSON",
+			useJSON:            true,
+			expectedHumanLabel: "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h",
+			expectedHumanDesc:  "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h: 1970-01-01T00:54:10Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: `
+        WITH cpu_avg AS (
+          SELECT time_bucket('3600 seconds', time) as hour, tags_id,
+          avg(usage_user) as mean_usage_user
+          FROM cpu
+          WHERE time >= '1970-01-01 00:54:10.138978 +0000' AND time < '1970-01-01 12:54:10.138978 +0000'
+          GROUP BY hour, tags_id
+        )
+        SELECT hour, tags->>'hostname', mean_usage_user
+        FROM cpu_avg
+        JOIN tags ON cpu_avg.tags_id = tags.id
+        ORDER BY hour, tags->>'hostname'`,
+		},
+		{
+			desc:               "use tags",
+			useTags:            true,
+			expectedHumanLabel: "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h",
+			expectedHumanDesc:  "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h: 1970-01-01T00:47:30Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: `
+        WITH cpu_avg AS (
+          SELECT time_bucket('3600 seconds', time) as hour, tags_id,
+          avg(usage_user) as mean_usage_user
+          FROM cpu
+          WHERE time >= '1970-01-01 00:47:30.894865 +0000' AND time < '1970-01-01 12:47:30.894865 +0000'
+          GROUP BY hour, tags_id
+        )
+        SELECT hour, tags.hostname, mean_usage_user
+        FROM cpu_avg
+        JOIN tags ON cpu_avg.tags_id = tags.id
+        ORDER BY hour, tags.hostname`,
+		},
+		{
+			desc:               "enable JSON and tags but use JSON",
+			useJSON:            true,
+			useTags:            true,
+			expectedHumanLabel: "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h",
+			expectedHumanDesc:  "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h: 1970-01-01T00:37:12Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: `
+        WITH cpu_avg AS (
+          SELECT time_bucket('3600 seconds', time) as hour, tags_id,
+          avg(usage_user) as mean_usage_user
+          FROM cpu
+          WHERE time >= '1970-01-01 00:37:12.342805 +0000' AND time < '1970-01-01 12:37:12.342805 +0000'
+          GROUP BY hour, tags_id
+        )
+        SELECT hour, tags->>'hostname', mean_usage_user
+        FROM cpu_avg
+        JOIN tags ON cpu_avg.tags_id = tags.id
+        ORDER BY hour, tags->>'hostname'`,
+		},
+	}
+
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(13 * time.Hour)
```

Comment:
Just to make this slightly more resilient, I think you should do it as something like:
```
e := s.Add(devops.DoubleGroupByDuration).Add(time.Hour)
```

Otherwise we might have subtle failures if that constant changes that are hard to track down. Same for other test funcs

Reply:
Fixed.